lib/CL/devices/basic/basic.c

```diff
@@ -293,6 +293,8 @@ pocl_basic_init_device_infos(unsigned j, struct _cl_device_id* dev)
 
   dev->device_side_printf = 1;
   dev->printf_buffer_size = 16 * 1024 * 1024;
+  dev->spir_compatible_triple = SPIR_HOST_TARGET_TRIPLE;
```

Comment:
I still don't understand the point of this. Now you seem to use it mainly as a boolean to check whether the target supports SPIR input. A bool makes more sense here as we need to force set for the target triple in any case to match the bitcode lib's target. Later on we can add a pass to ensure alignment is explicit in all loads and stores in the SPIR input, if they are not mandated to be.

Reply:
OK, then just 2 bools? ;) Still not seeing the point of storing a datalayout string here if we force set to the bitcode lib anyhow.

Reply:
Or just reuse the 32b property already available in cl_device properties.

Reply:
We're not storing the datalayout string here, we're storing a SPIR triple only, but i'll try to simplify it.

Reply:
Ah right. But yes, better have option "spir_supported" and "spirv_supported" and the bitness read from the ocl device's bitness?